src/ui/units/ql/utils/visualization/getQlVisualization.ts

```diff
@@ -0,0 +1,41 @@
+import mergeWith from 'lodash/mergeWith';
+import {Shared} from 'shared';
+
+import {getAvailableQlVisualizations, getDefaultQlVisualization} from '../visualization';
+
+export const getQlVisualization = (
+    loadedVisualization: Shared['visualization'],
+): Shared['visualization'] => {
+    const visualizationId =
+        loadedVisualization.id === 'table' ? 'flatTable' : loadedVisualization.id;
+    const availableVisualizations = getAvailableQlVisualizations();
+
+    const candidate = availableVisualizations.find((vis) => vis.id === visualizationId);
+
+    if (candidate) {
+        return mergeWith(
+            {},
+            candidate,
+            loadedVisualization,
+            (
+                value: any,
+                _sourceValue: any,
+                key,
+                obj: Shared['visualization'] | {},
+                source: Shared['visualization'],
+            ) => {
+                // it needs to prevent override properties which already defined in candidate visualization
```

Comment:
Please, start comments with capital leter
`// It needs to prevent override properties which already defined in candidate visualization`